d2ir/compile.go

```diff
@@ -59,13 +60,18 @@ func (c *compiler) compileSteps(m *Map) {
 		return
 	}
 	for i, sf := range steps.Fields {
-		if sf.Map() == nil {
+		if sf.Map() == nil || sf.Primary() != nil {
+			c.errorf(sf.References[0].Context.Key, "invalid step")
 			continue
 		}
 		var base *Map
 		if i == 0 {
 			base = m.CopyBase(sf)
 		} else {
+			if steps.Fields[i-1].Map() == nil || steps.Fields[i-1].Primary() != nil {
```

Comment:
Remove this and break above instead. This error isn't accurate anyway and reports the wrong step as being invalid.